std-bits/table/src/main/java/org/enso/table/data/column/storage/numeric/AbstractLongStorage.java

```diff
@@ -166,13 +207,94 @@ public AbstractLongStorage fillMissingFromPrevious(BoolStorage missingIndicator)
   public abstract AbstractLongStorage widen(IntegerType widerType);
 
   @Override
-  public long get(long index) throws ValueIsNothingException {
-    if (isNothing(index)) {
-      throw new ValueIsNothingException(index);
+  public Storage<Long> applyFilter(BitSet filterMask, int newLength) {
+    var builder =
+        Builder.getForLong(IntegerType.INT_64, newLength, BlackholeProblemAggregator.INSTANCE);
+    Context context = Context.getCurrent();
+    for (int i = 0; i < getSize(); i++) {
+      if (filterMask.get(i)) {
+        if (isNothing(i)) {
+          builder.appendNulls(1);
+        } else {
+          builder.appendLong(getItemAsLong(i));
+        }
+      }
+
+      context.safepoint();
+    }
+    return builder.seal();
+  }
+
+  @Override
+  public Storage<Long> applyMask(OrderMask mask) {
+    var builder =
+        Builder.getForLong(IntegerType.INT_64, mask.length(), BlackholeProblemAggregator.INSTANCE);
```

Comment:
Again, applying a mask should not widen the type but I suspect it may?

Reply:
a mistake will fix.